### DISCUSSION TITLE
SvelteKit signIn redirect: false is not working?

### DISCUSSION THREAD

**Discussion by nmicun** [nextauthjs/next-auth#6464] [Help] [answered]
Using email provider with magic code (token). After signIn I do not want to redirect on verify-request page before user inserts magic code. Im trying with this but its not working, still redirect on verify page in production, but in debug mode it works as expected:
`const res = signIn("email", {email: email, redirect: false})`

**Comment by canastro** [comment, marked as answer]
I'm facing the same issue, taking a quick peak into the code it seems like redirect is ignored if theres no error returned from the API:


```ts
  if (redirect || !isSupportingReturn || !error) {
    // TODO: Do not redirect for Credentials and Email providers by default in next major
    window.location.href = data.url ?? callbackUrl
    // If url contains a hash, the browser does not reload the page. We reload manually
    if (data.url.includes("#")) window.location.reload()
    return
  }
``` 
https://github.com/nextauthjs/next-auth/blob/main/packages/frameworks-sveltekit/src/lib/client.ts#L66-L71

**Reply by nmicun** [reply]
Yes, I  solved this by removing !error from if statement.

**Reply by canastro** [reply]
@nmicun this is code in the authjs/sveltekit, how did you fix it?
